lib/elixir/lib/protocol.ex

```diff
@@ -666,7 +666,7 @@ defmodule Protocol do
       end
 
     new_signatures =
-      for {{fun, arity}, :def, _, _} <- definitions do
+      for {{fun, arity}, :def, _, _} when arity > 0 <- definitions do
```

Comment:
Unfortunately this is not enough to fix it. If people are defining custom functions, then we need to do this modification only for protocol functions. So we need to store somewhere, either in a module attribute or perhaps in the function metadata, the protocol functions and traverse only them here.

Reply:
Maybe we can replace this line:

https://github.com/elixir-lang/elixir/blob/7a1a3ab32bafbd2dbd79dfcf59219d140e490bd5/lib/elixir/lib/protocol.ex#L292

By:

    Kernel.def(unquote({name, [protocol: true], args}))

And see if `[protocol: true]`appears in the metadata here.

---

Another option, which perhaps I like most, is to traverse the implementation of the `__protocol__` and find the return for the `:functions` clause: https://github.com/elixir-lang/elixir/blob/7a1a3ab32bafbd2dbd79dfcf59219d140e490bd5/lib/elixir/lib/protocol.ex#L974

Reply:
And then we need to add a test here that the signature for the user generated function does not have a `:strong` triplet:

https://github.com/elixir-lang/elixir/blob/7a1a3ab32bafbd2dbd79dfcf59219d140e490bd5/lib/elixir/test/elixir/protocol/consolidation_test.exs#L175

It should most likely be an `:infer` tuple (you can match  on `{:infer, _, _}`).

Reply:
> Another option, which perhaps I like most, is to traverse the implementation of the __protocol__ and find the return for the :functions clause: 

I like it, seems straightforward. This is what you mean right? [609f7cd](https://github.com/elixir-lang/elixir/pull/14191/commits/609f7cde69f924f1d23c2f8f70392e98a63dd0a8)

> And then we need to add a test here that the signature for the user generated function does not have a :strong triplet:

Will have a look!